src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpJupyterDAO.scala

```diff
@@ -8,18 +8,20 @@ import akka.pattern.ask
 import akka.stream.ActorMaterializer
 import akka.util.Timeout
 import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.leonardo.dns.{ClusterDnsCache, DnsCacheKey}
 import org.broadinstitute.dsde.workbench.leonardo.model.google.ClusterName
 import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
 import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
 
-class HttpJupyterDAO(val clusterDnsCache: ActorRef)(implicit system: ActorSystem, materializer: ActorMaterializer, executionContext: ExecutionContext) extends JupyterDAO with LazyLogging {
+class HttpJupyterDAO(val clusterDnsCache: ClusterDnsCache)(implicit system: ActorSystem, materializer: ActorMaterializer, executionContext: ExecutionContext) extends JupyterDAO with LazyLogging {
 
   val http = Http(system)
 
-  override def getStatus(googleProject: GoogleProject, clusterName: ClusterName): Future[Boolean] = {
+  override def isProxyAvailable(googleProject: GoogleProject, clusterName: ClusterName): Future[Boolean] = {
```

Comment:
Removed the `isProxyAvailable` method that used to be in `ClusterMonitorActor` and renamed `JupyterProxyDAO.getStatus` as `isProxyAvailable` and started calling that here directly because

1. with the call `ensureClusterReadyForProxying(ip, clusterStatus)` no longer necessary, this method would've been down to the `jupyterProxyDAO.getStatus` call
2. `JupyterProxyDAO.getStatus` was not returning us a `status` per se but whether the proxy is available or not